### PR TITLE
Fix notes plugin: avoid crashes on missing note definitions and improve lookup handling

### LINKED ISSUE #4763: `hls-notes-plugin` crashes instead of reporting no locations
The `hls-notes-plugin` reports an error instead of no locations when it can't find the note in question.

Rather, we should just show the same thing like when "Goto definition" fails.

Image

~~Slightly unrelated, I don't know why it fails to find the note, as it does exist, but that's another question.~~ The note `Note [Shutdown]` is ill-formatted.

### PR DESCRIPTION
**Pull request by SATVIKsynopsis**
Fixes #4763 

This PR updates the Notes plugin to safely handle cases where note definitions or references are missing.
So, Instead of throwing internal errors, the plugin now returns `Null` results as expected by the LSP specs.

Changes include:

1. Graceful handling of missing notes in jump-to-definition and references.

2. Replacing unsafe lookups with `fromMaybe`.

3. Cleaning up unused imports.

Functionality tested with a simple example;   jump-to-definition works correctly and missing notes no longer cause errors.

### CODE REVIEW

File: plugins/hls-notes-plugin/src/Ide/Plugin/Notes.hs [hunk 1]
```diff
@@ -100,25 +100,36 @@ findNotesRules recorder = do
...
+                Nothing -> pure Nothing
+                Just ln ->
+                    
+                    pure $
+                        listToMaybe $
+                            mapMaybe (atPos (fromIntegral c)) $
+                                matchAllText noteRefRegex ln
   where
-    atPos c arr = case arr A.! 0 of
-        -- We check if the line we are currently at contains a note
```
Comment (jvanbruegge) [inline plugins/hls-notes-plugin/src/Ide/Plugin/Notes.hs:116]: Why delete this comment?

File: plugins/hls-notes-plugin/src/Ide/Plugin/Notes.hs [hunk 2]
```diff
@@ -100,25 +100,36 @@ findNotesRules recorder = do
...
-    contents <-
-        err "Error getting file contents"
-        =<< liftIO (runAction "notes.getfileContents" state (getFileContents nfp))
-    line <- err "Line not found in file" (listToMaybe $ Rope.lines $ fst
-        (Rope.splitAtLine 1 $ snd $ Rope.splitAtLine (fromIntegral l) contents))
-    pure $ listToMaybe $ mapMaybe (atPos $ fromIntegral c) $ matchAllText noteRefRegex line
+    mContents <- liftIO (runAction "notes.getfileContents" state (getFileContents nfp))
+    case mContents of
+        Nothing ->
+
```
Comment (jvanbruegge) [inline plugins/hls-notes-plugin/src/Ide/Plugin/Notes.hs]: Why the extra newlines?

**Comment by SATVIKsynopsis** [comment]
Thanks for the feedback

1. I have updated the `listReferences` and `jumpToNote` handlers as requested

2. Removed the previous err-based handling for missing notes and references.

3. Added explicit case `HM.lookup` branches to return `Null` when the note or reference is not found.

4. Ensured both handlers now match the behavior of `“Goto Definition”` when no match exists.

5. No unrelated logic or formatting was changed.

Please let me know if anything else should be adjusted.

**Comment by mati865** [comment]
@jvanbruegge a heads-up, it might be hard to review this PR because this is at least largely a chatbot's work. You can see similar pull requests to other projects on author's profile.

**Comment by SATVIKsynopsis** [comment]
To clarify, I reviewed the changes myself before submitting the PR.
If anything looks off or needs adjustment I am m happy to revise it. Please let me know what should be improved.

**Comment by jvanbruegge** [comment]
The code looks now fine, but the pre-commit hook needs to pass and it should have a test for the new empty responses

**Comment by SATVIKsynopsis** [comment]
Thanks for the review

I am currently focusing on Rust contributions and I no longer have the Haskell setup locally.
Because of this, I won’t be able to continue this PR right now.

Feel free to close it or let someone else pick it up.
I learned from the review and appreciate your time.

**Comment by fendor** [comment]
Superseded by #4828
